fastly/ngwaf/v1/workspaces/api_test.go

```diff
@@ -0,0 +1,209 @@
+	}
+	if ws.AttackSignalThresholds.OneHour != 0 {
+		t.Errorf("unexpected workspace attack signal thresholds one_hour parameter: got %v, expected %v", ws.AttackSignalThresholds.OneHour, *wsAttackSignalThresholds.OneHour)
+	}
+
+	// Ensure we delete the test workspace at the end.
+	defer func() {
+		fastly.Record(t, "delete_workspace", func(c *fastly.Client) {
+			err = Delete(c, &DeleteInput{
+				WorkspaceID: fastly.ToPointer(ws.WorkspaceID),
+			})
+		})
+		if err != nil {
+			t.Errorf("error during workspace cleanup: %v", err)
+		}
+	}()
+
+	// Get the test workspace.
+	var gws *Workspace
+	fastly.Record(t, "get_workspace", func(c *fastly.Client) {
+		gws, err = Get(c, &GetInput{
+			WorkspaceID: fastly.ToPointer(ws.WorkspaceID),
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if gws.Name != ws.Name {
+		t.Errorf("unexpected workspace name: got %q, expected %q", gws.Name, ws.Name)
+	}
+	if gws.Description != ws.Description {
+		t.Errorf("unexpected workspace description: got %q, expected %q", gws.Description, ws.Description)
+	}
+	if gws.Mode != ws.Mode {
+		t.Errorf("unexpected workspace mode: got %q, expected %q", gws.Mode, ws.Mode)
+	}
+	if gws.IPAnonymization != ws.IPAnonymization {
+		t.Errorf("unexpected workspace IP anonymization: got %q, expected %q", gws.IPAnonymization, ws.IPAnonymization)
+	}
+	if gws.AttackSignalThresholds.Immediate != ws.AttackSignalThresholds.Immediate {
+		t.Errorf("unexpected workspace attack signal thresholds immediate parameter: got %t, expected %t", gws.AttackSignalThresholds.Immediate, ws.AttackSignalThresholds.Immediate)
+	}
+	if gws.AttackSignalThresholds.OneMinute != ws.AttackSignalThresholds.OneMinute {
+		t.Errorf("unexpected workspace attack signal thresholds one_minute parameter: got %v, expected %v", gws.AttackSignalThresholds.OneMinute, ws.AttackSignalThresholds.OneMinute)
+	}
+	if gws.AttackSignalThresholds.TenMinutes != ws.AttackSignalThresholds.TenMinutes {
+		t.Errorf("unexpected workspace attack signal thresholds ten_minutes parameter: got %v, expected %v", gws.AttackSignalThresholds.TenMinutes, ws.AttackSignalThresholds.TenMinutes)
+	}
+	if gws.AttackSignalThresholds.OneHour != ws.AttackSignalThresholds.OneHour {
+		t.Errorf("unexpected workspace attack signal thresholds one_hour parameter: got %v, expected %v", gws.AttackSignalThresholds.OneHour, ws.AttackSignalThresholds.OneHour)
+	}
+
+	// Update the test workspace.
+	const uwsName = "test-workspace"
+	const uwsDescription = "test-description"
+	const uwsMode = "log"
+	const uwsIPAnonymization = "hashed"
+
+	uwsAttackSignalThresholds := new(AttackSignalThresholdsInput)
+	uwsAttackSignalThresholds.OneMinute = fastly.ToPointer(10000)
+	uwsAttackSignalThresholds.TenMinutes = fastly.ToPointer(10000)
+	uwsAttackSignalThresholds.OneHour = fastly.ToPointer(10000)
+	uwsAttackSignalThresholds.Immediate = fastly.ToPointer(true)
```

Comment:
These are the same values as used during the 'create' operation, so once CDTOOL-1072 has been resolved these will need to be different values in order to ensure that the 'update' operation changed them.

Reply:
Good catch! Addressed in ec0bf8d76a45bd54ec3e290d34d8a4ba62579523.